### ISSUE TITLE
414 Request Too Large

### ISSUE THREAD

**Issue by gdinant** [Project-OSRM/osrm-backend#1875]
Hi there,

I actually need to keep internally some big distance matrix synch (~500x500) and I'm having a blocker with the current API since all parameters are passed through HTTP GET. Above 300 elements I reach 8kb and nginx rejects the request (414 Request Too Large) http://{server}/table?loc={lat,lon}&loc={lat,lon}<&loc={lat,lon} ...>.

Could it be possible to have another implementation accepting POST Requests taking a JSON body looking like:

```
[
  {
     lat : 51.3231,
     lon : 63.2314
  },
  {
     lat : 32.231,
     lon : 31.123
  }
]
```

or { x : ... , y : .... } ?

In summary :
 
```
HTTP POST http://{server}/table
[
  {
     lat : 51.3231,
     lon : 63.2314
  },
  {
     lat : 32.231,
     lon : 31.123
  }
]
```

Thanks

**Comment by daniel-j-h** [comment]
If it's just nginx that's making troubles, why not increasing its [client_max_body_size](http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size)?
If that's not possible, it should be fairly easy to split the distance table request into multiple sub-requests.

**Comment by gdinant** [comment]
This is a server config not a client one. On my side I can't do much. To me it seems fairly logical that big requests should be implemented with a body.

Yes I could do multiple requests but still more a workaround than a real solution to me.

**Comment by daniel-j-h** [comment]
I see --- I thought you were reverse-proxy'ing a osrm routed server with nginx.

**Comment by hoerup** [comment]
Doesn't osrm-routed support POST requests already ?

**Comment by gdinant** [comment]
If yes, I can't find it in the documentation.

**Comment by hoerup** [comment]
Yup:

# curl -v --data "src=55.2622189848261,9.71226630748429&dst=55.2585,9.70742" http://127.0.0.1:5000/table
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 5000 (#0)
> POST /table HTTP/1.1
> User-Agent: curl/7.38.0
> Host: 127.0.0.1:5000
> Accept: */*
> Content-Length: 57
> Content-Type: application/x-www-form-urlencoded
>
* upload completely sent off: 57 out of 57 bytes
* HTTP 1.0, assume close after body
< HTTP/1.0 200 OK
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET
< Access-Control-Allow-Headers: X-Requested-With, Content-Type
< Content-Length: 132
< Content-Type: application/json; charset=UTF-8
< Content-Disposition: inline; filename="response.json"
<
* Closing connection 0
{"status":200,"source_coordinates":[[55.262184,9.712255]],"destination_coordinates":[[55.258438,9.707142]],"distance_table":[[603]]}

**Comment by gdinant** [comment]
Well I'm getting 503 errors now ... not able to validate your query on the demo environment.

But based on your answer how do you implement a list of points with your current solution? Why not to stick to the response format and accept json instead?

**Comment by gdinant** [comment]
@hoerup Your solution works. Someone should update the API documentation accordingly.

**Comment by krashish8** [comment]
@hoerup @gdinant Is it still possible to use POST request with the OSRM API, just in case if you know?

My use case is similar: For now, I have to do a GET request on http://{server}/table/v1/driving/{lon1},{lat1};{lon2},{lat2};{lon3},{lat3}
But, if the number of coordinates is large, I get the "414 Request-URI Too Large" response.

If possible, can you please point out how do I get a similar response using a POST request, or writing the coordinates in the body of the request instead of the URL, so as to avoid the 414 Response?

**Comment by jcoupey** [comment]
> Is it still possible to use POST request with the OSRM API

No, support for POST has been removed [for a long time](https://github.com/Project-OSRM/osrm-backend/blob/master/CHANGELOG.md#500-rc1). You can either fiddle with your settings to change the default GET request length or use bindings rather than `osrm-routed`.

**Comment by danpat** [comment]
Also note that `osrm-routed` has no `GET` size limit - a `414` error response is either coming from some kind of proxy between your client and the `osrm-routed` server, or from the client code itself not supporting large `GET` requests.

**Comment by arbass22** [comment]
I am also running into this issue with large table requests!  Its seems like the issue is probably coming from the AWS load balancer we are using but unfortunately this limit is not configurable.  The ideal solution would be if OSRM supported post requests for these situations, but we can likely also break up the request into multiple and rejoin the responses in our client

**Comment by danpat** [comment]
Depending on how much energy you have, note that there is a version of `osrm-routed` ported to NodeJS/Express on this branch here:

https://github.com/Project-OSRM/osrm-backend/tree/osrm-routed.js

It re-implements most of the functionality from the C++ `osrm-routed` server in NodeJS.  It'd be pretty easy to add `POST` support to that if you need it.
